addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -786,36 +804,10 @@ export function createRelatedModels(modelDefs, modelClasses = {}, indexes = {})
             }
         }
 
-        for (const [model, values] of Object.entries(results)) {
-            const valuesToAdd = [];
-            const valuesToUpdate = [];
-
-            if (!(model in orderedRecords)) {
-                continue;
-            }
-
-            if (orderedRecords[model].length === 0) {
-                orderedRecords[model] = values;
-                valuesToAdd.push(...values);
-            } else {
-                for (const value of values) {
-                    const index = orderedRecords[model].findIndex((or) => or.id === value.id);
-
-                    if (index === -1) {
-                        valuesToAdd.push(value);
-                    } else {
-                        valuesToUpdate.push([index, value]);
-                    }
-                }
-
-                for (const [index, value] of valuesToUpdate) {
-                    orderedRecords[model][index] = value;
-                }
-                orderedRecords[model].unshift(...valuesToAdd);
-            }
-
-            const event = valuesToAdd.length > 0 ? "create" : "update";
-            models[model].triggerEvents(event, values);
+        for (const [model, values] of Object.entries(callbacks)) {
+            const modelInst = models[model];
+            values.created.forEach((values) => modelInst.triggerEvents("create", values));
+            values.updated.forEach((values) => modelInst.triggerEvents("update", values));
```

Comment:
`(values) => modelInst.triggerEvents("create", values)`

`values` in that callback isn't really an array of value, it's just single value.

Reply:
And the `callback` that is called by `triggerEvents` expects an array.